core/src/main/java/nl/nn/adapterframework/doc/objects/AClass.java

```diff
@@ -29,13 +37,101 @@
     private String javadocLink;
     private ArrayList<String> superClasses;
     private ArrayList<AMethod> methods;
+    private String referredClassName = "";
+
 
-    public AClass(String name, String packageName, String javadocLink, ArrayList<String> superClasses) {
+    public AClass(String name) {
         this.name = name;
-        this.packageName = packageName;
-        this.javadocLink = javadocLink;
-        this.superClasses = superClasses;
-        this.methods = new ArrayList<AMethod>();
+        this.superClasses = new ArrayList<>();
+        this.methods = new ArrayList<>();
+    }
+    /**
+     * Get the superclasses of a certain class.
+     *
+     * @param referredClassName - The class we have to derive the superclasses from
+     * @return An ArrayList containing all the superclasses with priority given to
+     *         them
+     */
+    public void setSuperclasses(String referredClassName) {
+        if (!referredClassName.isEmpty()) {
+            superClasses.add(referredClassName);
+        }
+        Class<?> clazz = null;
+        try {
+            clazz = Class.forName(packageName);
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+            System.out.println("["  + packageName + "] is not found");
+        }
```

Comment:
In geval van een fout moet je niet e.printStackTrace(); en System.out.println() doen, maar een warning of error loggen naar een logger.